design/Soft_finality_guarantee.md

```diff
@@ -0,0 +1,161 @@
+it breaks them.
+Basically, instead of being challenged over the result of executing transactions, the sequencer can be challenged over the root hash 
+of the list of L2 transactions alone.
+
+A sequencer will be able to generate sibling LBs in case an L1 fork is happening, but only as long as the order of the L2 
+transactions is preserved.
+
+```go
+type LightBatchHeader struct{
+	ParentLB        Hash    // the id of the parent Light Batch
+	L1BlockHash     Hash    // the head l1 block at the time of this LB
+	L2TxsHash       Hash    // the MTree root of the list of transactions
+	Number          int     // the height of this LB - since when? the latest L1 block, the latest rollup, the beginning?
+	StateRoot       Hash    // the state root hash
+	R, S            *big.Int // signature of enclave
+}
+```
+
+There can be sibling LBs (same `Number` and `ParentLB`) with the same `L2TxsHash` but different `L1BlockHash`. 
+
+This option offers a much stronger finality guarantee. An L2 transaction can never be replaced with a competing one, 
+but it might result in a different outcome only in the case a deposit L1 transaction was processed or not before it.
+
+The sibling LBs will result in the same execution result for the vast majority of transactions. 
+
+If a user is not actively trying to double spend a deposit on the L1, then the odds that an L1 re-org have an impact
+on the result of executing their transaction are minimal. 
+
+
+### Option 4 - Guarantee the L2 ordering, but link batches to deposit messages
+
+Another possible improvement is to reduce the reasons for creating sibling LBs to only the situation where
+there is an actual change in the ordering of L1 messages (deposits).
+
+The insight is that instead of linking the batch to an L1 block, we can link a batch to a list of L1 messages.
+If the order of L1 messages is preserved during an L1 reorg ( which is very likely unless the sender actively
+tries an attack), then there is no reason for generating a sibling LB.
+
+```go
+type LightBatchHeader struct{
+	ParentLB        Hash    // the id of the parent Light Batch
+	L1Number        int     // the height of the l1 block used 
+	L1MsgsHash      Hash    // the list of messages from the L1.
+	L2TxsHash       Hash    // the MTree root of the list of transactions
+	Number          int     // the height of this LB - since when? the latest L1 block, the latest rollup, the beginning?
+	StateRoot       Hash    // the state root hash
+	R, S            *big.Int // signature of enclave
+}
+```
+
+The drawback of this solution is some logical complexity. As the validators process blocks and batches, they might reach different L1MsgsHash, which needs handling.
+
+Same as in Option 3, an L2 transaction can never be replaced with a competing one, but it might result in a different outcome only if there is a deliberate
+double-spend attack on the L1 by the same account.  
+
+## Conclusion
+
+The last two options provide the best UX.
```

Comment:
Can you explain whether it's possible to start with option 3, then move to option 4?